### PR TITLE
Incorporate useReducer and useContext to mange Feed state

### PR DESCRIPTION
**Pull request by anniegiang**

### CODE REVIEW

File: client/src/reducers/feedReducers.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  TOGGLE_STATE,
+  SET_VALUE,
+} from "../actions/feedActions";
+
+export const feedReducer = (oldState, action) => {
+  let newState = Object.assign({}, oldState);
+  const { type, key, value } = action;
+  switch (type) {
+    case TOGGLE_STATE:
+      newState[key] = !newState[key];
```
Comment (robinv85) [inline client/src/reducers/feedReducers.js:14]: I would probably write it a little shorter too like `return { ...state, [key]: !state[key] };`

File: client/src/reducers/feedReducers.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  }
+};
+
+export const optionsReducer = (oldState, action) => {
+  const { option, label } = action.payload;
+  let newState = Object.assign({}, oldState);
+  newState[label] = newState[label] || [];
+  switch (action.type) {
+    case ADD_OPTION:
+      newState[label].push(option);
```
Comment (robinv85) [inline client/src/reducers/feedReducers.js:30]: if you clone the `oldState` here and it keeps reference to the `oldState[label]` array your should also immutably update the nested object and arrays, so not using `[].push()` but e.g. 
```
return { ...oldState, [label]: [ ...oldState[label], option ] };
```
more examples here https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns

**Comment by anniegiang** [comment]
@robinv85 made the changes!

**Comment by robinv85** [comment]
Ok 👌🏼

**Pull request merged**